engine/polyglot-api/src/main/scala/org/enso/polyglot/Suggestion.scala

```diff
@@ -330,6 +333,78 @@ object Suggestion {
       s",reexport=$reexport)"
   }
 
+  /** Base trait for method suggestions. */
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+  @JsonSubTypes(
+    Array(
+      new JsonSubTypes.Type(
+        value = classOf[Suggestion.Getter],
+        name  = "suggestionMethodGetter"
+      ),
+      new JsonSubTypes.Type(
+        value = classOf[Suggestion.DefinedMethod],
+        name  = "suggestionMethodDefinedMethod"
+      ),
+      new JsonSubTypes.Type(
+        value = classOf[Suggestion.Conversion],
+        name  = "suggestionMethodConversion"
+      )
+    )
+  )
+  sealed trait Method extends Suggestion {
+    def arguments:   Seq[Argument]
+    def selfType:    String
+    def isStatic:    Boolean
+    def annotations: Seq[String]
+    def reexport:    Option[String]
+  }
+
+  /** A method generated to access constructor field.
+    *
+    * @param externalId the external id
+    * @param module the module name
+    * @param name the method name
+    * @param arguments the list of arguments
+    * @param selfType the self type of a method
+    * @param returnType the return type of a method
+    * @param documentation the documentation string
+    * @param annotations the list of annotations
+    * @param reexport the module re-exporting this method
+    */
+  case class Getter(
+    externalId: Option[ExternalId],
+    module: String,
+    name: String,
+    arguments: Seq[Argument],
```

Comment:
Can a getter have non-empty list arguments?

Reply:
You're right, I should remove the `arguments` parameter.

Reply:
Silly me. Actually getters does have a `self` argument.